cmd/vic-machine/common/dns.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"net"
+
+	log "github.com/Sirupsen/logrus"
+	"gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/vic/pkg/errors"
+)
+
+// general dns
+type DNS struct {
+	dns   cli.StringSlice `arg:"dns-server"`
+	IsSet bool
+}
+
+func (d *DNS) DNSFlags(hidden bool) []cli.Flag {
+	return []cli.Flag{
+		cli.StringSliceFlag{
+			Name:   "dns-server",
+			Value:  &d.dns,
+			Usage:  "DNS server for the client, public, and management networks. Defaults to 8.8.8.8 and 8.8.4.4 when VCH uses static IP",
+			Hidden: hidden,
+		},
+	}
+}
+
+// processDNSServers parses DNS servers used for client, public, mgmt networks
+func (d *DNS) ProcessDNSServers() ([]net.IP, error) {
+	var parsedDNS []net.IP
+	if len(d.dns) > 0 {
+		d.IsSet = true
+	}
+
+	for _, n := range d.dns {
+		if n != "" {
+			s := net.ParseIP(n)
+			if s == nil {
+				return nil, errors.New("Invalid DNS server specified")
+			}
+			parsedDNS = append(parsedDNS, s)
+		}
+	}
+
+	// user input empty string, reset dns-server to default
+	if d.IsSet && parsedDNS == nil {
+		return []net.IP{net.ParseIP("8.8.8.8"), net.ParseIP("8.8.4.4")}, nil
```

Comment:
This should only be the case if DHCP is not being used. See https://github.com/vmware/vic/blob/master/lib/install/validate/network.go#L237 for what happens during create.